demisto_sdk/commands/content_graph/objects/base_content.py

```diff
@@ -160,32 +165,102 @@ def to_dict(self) -> Dict[str, Any]:
         if "path" in json_dct and Path(json_dct["path"]).is_absolute():
             json_dct["path"] = (Path(json_dct["path"]).relative_to(CONTENT_PATH)).as_posix()  # type: ignore
         json_dct["content_type"] = self.content_type
-
         return json_dct
 
+    def add_relationship(
+        self, relationship_type: RelationshipType, relationship: "RelationshipData"
+    ) -> None:
+        if relationship.content_item_to == self:
+            # skip adding circular dependency
+            return
+        self.relationships_data[relationship_type].add(relationship)
+
+
+class BaseContentWithPath(BaseContent):
```

Comment:
BaseContent: the base class of all the nodes - Packs, Content Items, Commands
BaseContentWithPath: the base class of Packs and Content Items. Because they have `path` they share a lot of behavior